Documentation/Design/.NET Tool v2.md

```diff
@@ -0,0 +1,71 @@
+# Functional design .NET tool 2.0 `coverlet`
+
+Tracking issue https://github.com/tonerdo/coverlet/issues/683
+
+The idea is to support some new verbs:
+
+# coverlet run
+
+This will support current instrumentation workflow
+
+```
+coverlet run /path/to/test-assembly.dll --target "dotnet" --targetargs "test /path/to/test-project --no-build"
```

Comment:
Suggestion: also add `--coverlet-batch ID` and save the ID in the JSON coverage file. Then the same `--coverlet-batch ID` can be provided to `mergereports` and `validate` to select which files to include, if the result directory isn't cleared between runs.

Reply:
My ideas was not "touch" old feature, to avoid new behaviour for current user...I mean we could add new feature to current .net tool, but it wasn't a goal of this tool.
Again...this run tool today will output a formant different than our json...so where we put that id?

Reply:
My comment was assuming the output here is coverlet json. Having this flag, which only would work if also selecting that format, would allow people to build a script that runs a series of specific test projects and then merges/validates the results based on the batch ID. (Though, in such a script, it's not too difficult to clean out the result dir or work in a temporary dir to ensure that only current report files are included)